src/tests/basic_pub_sub.rs

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+use crate::pb::{self, POD};
+use crate::sb;
+use crate::Runtime;
+use crate::testing::RouDiEnvironment;
+
+#[repr(C)]
+struct CounterTopic {
+    counter: u32,
+}
+
+unsafe impl POD for CounterTopic {}
+
+#[test]
+fn basic_pub_sub() {
+    let _roudi = RouDiEnvironment::new();
+
+    Runtime::init("publisher_simple");
+
+    let topic = sb::TopicBuilder::<CounterTopic>::new("Radar", "FrontLeft", "Counter")
+        .queue_capacity(5)
+        .build();
+
+    let (subscriber, sample_receive_token) = topic.subscribe();
+
+    let topic = pb::TopicBuilder::<CounterTopic>::new("Radar", "FrontLeft", "Counter").build().expect("Topic");//?;
+
+    let publisher = topic.offer();
+    let mut sample = publisher.allocate_sample().expect("Samle");//?;
+    sample.counter = 42;
+    publisher.publish(sample);
+
+    println!("Sending: {}", 42);
+
+    let sample_receiver = subscriber.get_sample_receiver(sample_receive_token);
+
```

Comment:
We should add some checks to make the test really work.
```suggestion
    assert_eq!(sample_receiver.has_samples(), true);
```